lsp/nls/src/requests/completion.rs

```diff
@@ -331,9 +328,15 @@ fn collect_record_info(
                 // Get record fields from static type info
                 (_, Types(TypeF::Record(rrows))) => find_fields_from_type(&rrows, path),
                 (TermKind::Declaration(_, _, ValueState::Known(body_id)), _) => {
-                    find_fields_from_contract(linearization, *body_id, path)
-                        .or_else(|| find_fields_from_term_kind(linearization, id, path))
-                        .unwrap_or_default()
+                    // The path is mutable, so the first case would consume the path
+                    // so we have to clone it so that it can be correctly used for the second case.
+                    let mut p = path.clone();
+                    let mut fst = find_fields_from_contract(linearization, *body_id, path)
+                        .unwrap_or_default();
+                    let snd =
+                        find_fields_from_term_kind(linearization, id, &mut p).unwrap_or_default();
```

Comment:
Then, does it make sense anymore to have an option returned by `find_fields_from` if we're aggregating? Would an empty collection be enough and make sense? (getting wrap of the `unwrap_or_default`)

Reply:
Yes, that would be much better. Thank you!

Reply:
But on a second thought, most of these functions use the monadic `?` operator on `Options` to make error propagation or early return much cleaner

Reply:
But do they need that at all? Couldn't you just return an empty collection instead?

Reply:
In any case, returning an option should be decided first and foremost by the semantics of the function and the callers though